functions-and-scope.js

```diff
@@ -30,54 +66,118 @@ const grades = [9, 8, 5, 7, 7, 4, 9, 8, 8, 3, 6, 8, 5, 6];
+// * Hoe zorgt ik ervoor dat ik alle waardes uit de array kan langslopen,
+//   ook als de array wel 100 entries zou bevatten?
 // Log het antwoord in de terminal.
 
+let sum = 0;
+for (let i = 0; i < grades.length; i++ ) {
+    sum = sum + grades[i];
+}
+console.log(sum / grades.length);
+
+
 // ---- Verwachte uitkomst: 6.642857142857143
 
 
 /* 2b: Omschrijven tot een herbruikbare functie */
-// Schrijf een functie genaamd averageGrade, die een array van cijfers verwacht (zoals grades) en het gemiddelde cijfer teruggeeft. Gebruik hiervoor jouw antwoord van 2a.
-// Zorg ervoor dat jouw functie ook werkt als we een andere array willen checken, zoals bijvoorbeeld: [6, 4, 5] of [8, 9, 4, 6, 10].
+// Schrijf een functie genaamd averageGrade, die een array van cijfers verwacht
+// (zoals grades) en het gemiddelde cijfer teruggeeft. Gebruik hiervoor jouw antwoord van 2a.
+// Zorg ervoor dat jouw functie ook werkt als we een andere array willen checken,
+// zoals bijvoorbeeld: [6, 4, 5] of [8, 9, 4, 6, 10].
 // Log het antwoord in de terminal.
 
+function averageGrade(arrayOfNumbers) {
+    let sum = 0;
+    for (let i = 0; i < arrayOfNumbers.length; i++) {
+            sum = sum + arrayOfNumbers[i];
+    }
+    return sum / arrayOfNumbers.length;
+}
+
+const average = averageGrade(grades);
+const average1 = averageGrade([6, 4, 5])
+const average2 = averageGrade([8, 9, 4, 6, 10])
+
+console.log(average);
+console.log(average1);
+console.log(average2);
+
+
 // ---- Verwachte uitkomsten:
 // averageGrade(grades) geeft 6.642857142857143
 // averageGrade([6, 4, 5]) geeft xxxx
 // averageGrade([8, 9, 4, 6, 10]) geeft xxxx
 
 
 /* 2c: Afronden op twee decimalen */
-// Zorg ervoor dat het gemiddelde cijfer dat wordt teruggegeven uit de functie netjes wordt afgerond op twee decimalen.
+// Zorg ervoor dat het gemiddelde cijfer dat wordt teruggegeven
+// uit de functie netjes wordt afgerond op twee decimalen.
 // Tip: Google is your best friend!
 
+const roundOfAverage = averageGrade(grades);
 
+console.log(roundOfAverage.toFixed(2));
 
 
 /* Bonusopdracht: hoogste cijfer */
 
 /* 3a: Script schrijven  */
-// Schrijf een script die op basis van de grades array (hierboven) checkt wat het hoogst behaalde cijfer is. Je mag hier geen bestaande methoden voor gebruiken. Schrijf de stapjes eerst uit en vraag jezelf de volgende dingen af:
+// Schrijf een script die op basis van de grades array (hierboven)
+// checkt wat het hoogst behaalde cijfer is. Je mag hier geen bestaande methoden voor gebruiken.
+// Schrijf de stapjes eerst uit en vraag jezelf de volgende dingen af:
 // * Hoe kan ik iedere waarde van de array langsgaan?
 // * Op welke conditie moet ik checken?
-// * Hoe zorgt ik ervoor dat wanneer ik een cijfer tegenkom die aan de conditie voldoet, ik dit ergens kan opslaan?
+// * Hoe zorgt ik ervoor dat wanneer ik een cijfer tegenkom die aan de conditie voldoet,
+//   ik dit ergens kan opslaan?
 // Log het antwoord in de terminal.
 
+let highestGrade = 0;
```

Comment:
Je kan hier beter niet 0 maar de eerste waarde uit de array pakken. Het zal hier altijd goed gaan, omdat in de praktijk cijfers niet lager dan 0 kunnen zijn. Maar stel je voor dat je een array aanlevert van [-1, -2, -5, -3], dan krijg je 0 terug wat dus niet klopt.